src/plugins/intel_gpu/src/kernel_selector/cl_kernels/sdpa_opt.cl

```diff
@@ -1247,55 +1272,142 @@ MAKE_VECTOR_TYPE(INPUT0_TYPE, TARGET_SEQ_LEN_BLOCK_SIZE) qk_acc = INPUT0_VAL_ZER
+                            const uint partition_idx = start_partition_idx / SEQ_LEN_PARTITION_SIZE;
+
+                            SOFTMAX_ACCUMULATOR_TYPE correction_factor = native_exp(qk_max_new - qk_max_cur);
+
+                            if (sglid == 0) {
+                                const uint max_partitions_num = aligned_max_context_len / SEQ_LEN_PARTITION_SIZE;
+                                const uint exp_sums_output_offset = subsequence_idx * NUM_HEADS * max_partitions_num +
+                                                                    num_heads_dim * max_partitions_num +
+                                                                    partition_idx;
+                                exp_sums[exp_sums_output_offset] = exp_sum_new * correction_factor;
+                                max_logits[exp_sums_output_offset] = qk_max_cur;
+                            }
+
+                            const uint output_offset = subsequence_idx * NUM_HEADS * aligned_max_context_len +
+                                                    num_heads_dim * aligned_max_context_len +
+                                                    partition_idx * SEQ_LEN_PARTITION_SIZE;
+                            for (uint i = sglid; i < partition_seq_len; i += SUBGROUP_SIZE) {
+                                softmax_results[output_offset + i] = TO_SOFTMAX_ACCUMULATOR_TYPE(slm_qk_vals[m][i]) / exp_sum_new;
+                            }
+                        }
+                    }
+                }
+#endif /*end of PAGED_ATTENTION_SCORES_OUTPUT*/
+
+                // update
+                if (sglid == 0) {
+                    SOFTMAX_ACCUMULATOR_TYPE pre_exp_sum = slm_exp_sum_prev[m];
+                    SOFTMAX_ACCUMULATOR_TYPE correction_factor = native_exp(max_val_prev - qk_max_new);
+                    SOFTMAX_ACCUMULATOR_TYPE pre_exp_sum_fixed = pre_exp_sum * correction_factor;
+                    exp_sum_new += pre_exp_sum_fixed;
+
+                    slm_update_factor[m] = correction_factor;
+                    slm_max_val_prev[m] = qk_max_new;
+                    slm_exp_sum_prev[m] = exp_sum_new;
+                }
+            }
+        }
+#else /*!IS_FLASHATTEN_V2*/
+        {
+            SOFTMAX_ACCUMULATOR_TYPE qk_max_new = SOFTMAX_ACCUMULATOR_VAL_MIN;
+
+            for (uint i = 0; i < SUBGROUPS_PER_WG; i++) {
+                SOFTMAX_ACCUMULATOR_TYPE qk_max_val = slm_qk_max_vals[i][sglid];
+                qk_max_new = SOFTMAX_ACCUMULATOR_MAX_FUNC(qk_max_new, qk_max_val);
+            }
+
+            if (sgid == 0) {
+                slm_max_val_cur[sglid] = qk_max_new;
+            }
+
+            SOFTMAX_ACCUMULATOR_TYPE exp_sum_new = SOFTMAX_ACCUMULATOR_VAL_ZERO;
+
+            for (uint i = 0; i < TARGET_SEQ_LEN_BLOCK_SIZE; i++) {
+                qk_acc[i] = native_exp(TO_SOFTMAX_ACCUMULATOR_TYPE(qk_acc[i]) - qk_max_new);
+#if IS_CAUSAL
+                if (seq_len + i <= target_seq_idx + sglid) {
```

Comment:
@sshlyapn  Do we need similar check macros like here - https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_gpu/src/kernel_selector/cl_kernels/sdpa_opt.cl#L1181 ? Any PA test case?

Reply:
Good point, we should handle this too. However, I think we can just force using `could_use_flashattn_v2=true` for all PA cases for now
There is no tests for sliding_window as far as I remember

Reply:
done